crml/nft/src/lib.rs

```diff
@@ -302,10 +344,195 @@ decl_module! {
+			let _ = Self::mint_series(origin, collection_id, One::one(), owner, attributes, metadata_path, royalties_schedule, None)?;
 			IsSingleIssue::insert(collection_id, series_id, true);
 		}
 
+		/// Buy tokens from an active mass_drop
+		/// Checks to make sure the drop has started
+		/// If the whitelist is not empty, only whitelisted accounts will be able to join
+		#[weight = 100000]
+		#[transactional]
+		fn enter_mass_drop(
+			origin,
+			collection_id: CollectionId,
+			series_id: SeriesId,
+			quantity: TokenCount,
+		) -> DispatchResult {
+			let origin = ensure_signed(origin)?;
+			let owner = Self::collection_owner(collection_id);
+			ensure!(owner.is_some(), Error::<T>::NoCollectionOwner);
+			let owner = owner.unwrap();
+			let is_owner = owner == origin.clone();
+			let serial_number = Self::next_serial_number(collection_id, series_id);
+			ensure!(
+				serial_number.checked_add(quantity).is_some(),
+				Error::<T>::NoAvailableIds
+			);
+			let mass_drop: MassDrop<T::AccountId> = Self::series_mass_drops(collection_id, series_id)
+				.ok_or(Error::<T>::NoMassDropExists)?;
+			let current_block = <frame_system::Module<T>>::block_number();
+
+			if current_block >= T::BlockNumber::from(mass_drop.activation_time) || is_owner{
+				// MassDrop has started, proceed with mint
+				if mass_drop.whitelist != vec![] {
+					ensure!(mass_drop.whitelist.contains(&origin) || is_owner, Error::<T>::NotInWhitelist);
+				}
+				if let Some(transaction_limit) = mass_drop.transaction_limit {
+					ensure!(quantity <= transaction_limit, Error::<T>::PurchaseQuantityTooHigh);
+				}
+				ensure!(serial_number + quantity <= mass_drop.max_supply, Error::<T>::NotEnoughTokens);
+				Self::process_drop_payment(&origin, owner, mass_drop.price, quantity.clone(), mass_drop.asset_id)?;
+			}else {
+				// Check if the presale has started
+				let pre_sale: PreSale<T::AccountId> = mass_drop.pre_sale
+					.ok_or(Error::<T>::MassDropNotStarted)?;
+				ensure!(current_block >= T::BlockNumber::from(pre_sale.activation_time), Error::<T>::PresaleNotStarted);
+				if pre_sale.whitelist != vec![] {
+					ensure!(pre_sale.whitelist.contains(&origin) || is_owner, Error::<T>::NotInWhitelist);
+				}
+				if let Some(transaction_limit) = pre_sale.transaction_limit {
+					ensure!(quantity <= transaction_limit, Error::<T>::PurchaseQuantityTooHigh);
+				}
+				ensure!(serial_number + quantity <= pre_sale.max_supply, Error::<T>::NotEnoughTokens);
+				Self::process_drop_payment(&origin, owner, pre_sale.price, quantity.clone(), mass_drop.asset_id)?;
+			}
+			Self::do_mint_additional(&origin, collection_id, series_id, serial_number, quantity)?;
+			Self::deposit_event(RawEvent::EnterMassDrop(collection_id, series_id, quantity, origin));
+			Ok(())
+		}
+
+		/// Change the activation time of an existing mass_drop
+		#[weight = 0]
+		#[transactional]
+		fn update_mass_drop_activation_time(
```

Comment:
```suggestion
		fn set_mass_drop_activation_time(
```
for consistency with set_whitelist